app/gui2/src/util/reactivity.ts

```diff
@@ -9,3 +20,220 @@ export function watchSourceToRef<T>(src: WatchSource<T>): Ref<T> {
 export function evalWatchSource<T>(src: WatchSource<T>): T {
   return isRef(src) ? src.value : src()
 }
+
+export type OnCleanup = (fn: () => void) => void
+export type StopEffect = () => void
+
+/**
+ * A set of effects that will defer their re-execution until an explicit flush. This is useful for
+ * implementing incremental updates to an external data structure, while delaying the update logic
+ * until next time that data structure is queried.
+ */
+export class LazySyncEffectSet {
+  dirtyRunners = new Set<() => void>()
+  scope = effectScope()
```

Comment:
@Frizi, these fields being private do not work with `pinia`, it seems like. Let’s have a call tomorrow.